trie/trie.go

```diff
@@ -245,14 +245,16 @@ func (t *Trie) tryGetWithEpoch(origNode node, key []byte, pos int, epoch types.S
 		}
 
 		if updateEpoch {
-			n.setEpoch(t.currentEpoch)
 			value, newnode, didResolve, err = t.tryGetWithEpoch(n.Val, key, pos+len(n.Key), t.currentEpoch, true)
 		} else {
 			value, newnode, didResolve, err = t.tryGetWithEpoch(n.Val, key, pos+len(n.Key), epoch, false)
 		}
-		if err == nil && didResolve {
+		if err == nil && didResolve || (updateEpoch && epoch != t.currentEpoch) {
```

Comment:
may using `err == nil && (didResolve || (updateEpoch && epoch != t.currentEpoch))` is safe, or using a function